network/index.md

```diff
@@ -104,6 +103,30 @@ You can override these settings on a per-container basis.
 | `--dns-opt`    | A key-value pair representing a DNS option and its value. See your operating system's documentation for `resolv.conf` for valid options.                                                                                                                            |
 | `--hostname`   | The hostname a container uses for itself. Defaults to the container's ID if not specified.                                                                                                                                                                          |
 
+### Name resolution with multiple nameservers
+
+When you specify multiple DNS servers using `--dns` flags, name resolution may
+work in a surprising or unexpected way. DNS lookup behavior depends on a number
+of different factors:
+
+- Whether the container OS runs on [musl or glibc](https://wiki.musl-libc.org/functional-differences-from-glibc.html#Name_Resolver/DNS){: target="blank" rel="noopener" }
+- Whether the Docker daemon binary was [statically or dynamically linked](https://pkg.go.dev/net#hdr-Name_Resolution){: target="blank" rel="noopener" }
+- If dynamically linked, which version of glibc that's used
+- Whether or not [nsswitch.conf is present](https://tldp.org/LDP/nag2/x-087-2-resolv.library.html#X-087-2-RESOLV.NSSWITCH-CONF){: target="blank" rel="noopener" }
+
+You may find that name resolution works as follows:
+
+1. The container emits requests to **all** nameservers that you specify.
+2. The container uses the first response returned by any of the nameservers.
+   Even if the first response is `NXDOMAIN`, or similar.
+
```

Comment:
I think we should also specify that currently any DNS server with an IPv6 address specified in the host `/etc/resolv.conf` will end up in the containers' `/etc/resolv.conf`. In practice, this means that container hostnames might sporadically fail if the external IPv6 DNS server win the race against the embedded DNS server.

As this might be a bit confusing, we should state that nonetheless IPv6 DNS records (ie. AAAA) are properly handled by the embedded DNS server.

For the record, here's the issue reporting that: https://github.com/moby/moby/issues/41651.